p4include/v1model.p4

```diff
@@ -88,27 +88,116 @@ enum MeterType {
 
 extern counter {
     counter(bit<32> size, CounterType type);
+    /***
+     * count() causes the counter state with the specified index to be
+     * read, modified, and written back, atomically relative to the
+     * processing of other packets, updating the packet count, byte
+     * count, or both, depending upon the CounterType of the counter
+     * instance used when it was constructed.
+     *
+     * @param index The index of the counter state in the array to be
+     *              updated, normally a value in the range [0,
+     *              size-1].  If index >= size, no counter state will be
+     *              updated.
+     */
     void count(in bit<32> index);
 }
 
 extern direct_counter {
     direct_counter(CounterType type);
+    /***
+     * count() on a direct_counter object may only be called within an
+     * action executed as a result of matching a table entry, of a
+     * table that has a direct_counter associated with it.  Calling
+     * count() causes the counter state associated with the matched
+     * table entry to be read, modified, and written back, atomically
+     * relative to the processing of other packets.
+     */
     void count();
 }
 
 extern meter {
     meter(bit<32> size, MeterType type);
+    /***
+     * execute_meter() causes the meter state with the specified index
+     * to be read, modified, and written back, atomically relative to
+     * the processing of other packets, and an integer encoding of one
+     * of the colors green, yellow, or red to be written to the result
+     * out parameter.
+     *
+     * @param index The index of the meter state in the array to be
+     *              updated, normally a value in the range [0,
+     *              size-1].  If index >= size, no meter state will be
+     *              updated.
+     * @param result Type T must be bit<W> with W >= 2.  When index is
+     *              in range, the value of result will be assigned 0
+     *              for color GREEN, 1 for color YELLOW, and 2 for
```

Comment:
@antoninbas Hoping you can do a quick double check on the values I documented here for GREEN, YELLOW, and RED.

Reply:
The bmv2 meter implementation doesn't restrict how many buckets / colors you can have, but yes, for a 2rTCM meter, the values are correct.

Reply:
Are you OK with leaving that generality of bmv2's implementation out of these comments?  I could mention it briefly if you want, but am afraid it is fairly far off the beaten track of how most people would use meters, and may only confuse some people.